examples/machine_learning/geneticalgorithm.cpp

```diff
@@ -0,0 +1,202 @@
+
+    array children_x2 = (parentsx2 & uppermask) + (parentsx1 & lowermask);
+    array children_y2 = (parentsy2 & uppermask) + (parentsy1 & lowermask);
+
+    //Join two new sets
+    samplex = join(0, children_x1, children_x2);
+    sampley = join(0, children_y1, children_y2);
+
+    //Create mutant children
+    array mutantx = samplex.copy();
+    array mutanty = sampley.copy();
+
+    //Flip a random bit to vary the gene pool
+    int bits = (int)(log((double)n) / log(2.0));
+    mutantx = mutantx ^ (pow2((randu(nsamples / 2, u32) % bits).as(f32))).as(u32);
+    mutanty = mutanty ^ (pow2((randu(nsamples / 2, u32) % bits).as(f32))).as(u32);
+
+    samplex = join(0, samplex, mutantx);
+    sampley = join(0, sampley, mutanty);
+
+    //Update the value of each sample with the new coordinates
+    update();
+}
+
+
+void init_samples() {
+    setSeed(time(NULL));
+    samplex = randu(nsamples, u32) % n;
+    sampley = randu(nsamples, u32) % n;
+    update();
+}
+
+
+void init() {
+    //initialize space
+    searchSpace = tile(array(seq(n / 2)), 1, n / 2) + tile(moddims(array(seq(n / 2)), 1, n / 2), n / 2);
+
+    searchSpace = join(0, searchSpace, flip(searchSpace, 0));
+    searchSpace = join(1, searchSpace, flip(searchSpace, 1));
+
+    //initialize display data
+    searchSpace_x_display = iota(dim4(n, 1), dim4(1, n));
+    searchSpace_y_display = iota(dim4(1, n), dim4(n, 1));
+
+    //initalize searchers
+    init_samples();
+}
+
+void geneticSearch(bool console) {
+
+    af::info();
+    init();
+
+    float truemax = max<float>(searchSpace);
+    bool found_max = false;
+
+    if (!console) {
+        af::Window win(1600, 800, "ArrayFire Genetic Algorithm Search Demo");
+        win.grid(1, 2);
+
+        do{
+            float maximum = max<float>(samplez);
+            array where_m = where(samplez == maximum);
+
+            if (maximum < truemax * 0.99) {
+                reproduce();
+                printf("Current max at (%d,%d): %f (truemax %f)\n",
+                        samplex(where_m).scalar<unsigned int>(),
+                        sampley(where_m).scalar<unsigned int>(), maximum, truemax);
+            } else if (!found_max) {
+                found_max = true;
+                printf("Max found at (%d,%d): %f (truemax %f)\n",
+                        samplex(where_m).scalar<unsigned int>(),
+                        sampley(where_m).scalar<unsigned int>(), maximum, truemax);
+            }
+
+            win(0,0).setAxesLimits(searchSpace_x_display, searchSpace_y_display, searchSpace);
+            win(0,0).setAxesTitles("idx", "idy", "Search Space");
+            win(0,1).setAxesLimits(searchSpace_x_display, searchSpace_y_display, searchSpace);
+            win(0,1).setAxesTitles("idx", "idy", "Search Space");
+            win(0,0).surface(searchSpace_x_display, searchSpace_y_display, searchSpace);
+            win(0,1).scatter(samplex.as(f32), sampley.as(f32), samplez.as(f32));
+            win.show();
```

Comment:
Just FYI, setAxesLimits has been reverted to automatic detection of limits in case you want to use that. You can test it by removing the `setAxesLimit` lines.